velox/exec/ValueStream.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/core/PlanNode.h"
+#include "velox/exec/Operator.h"
+
+namespace facebook::velox::exec {
+
+class ValueStream : public SourceOperator {
+ public:
+  ValueStream(
+      int32_t operatorId,
+      DriverCtx* driverCtx,
+      std::shared_ptr<const core::ValueStreamNode> ValueStreamNode);
+
+  RowVectorPtr getOutput() override;
+
+  BlockingReason isBlocked(ContinueFuture* /* unused */) override {
+    return BlockingReason::kNotBlocked;
+  }
+
+  void noMoreInput() override {
+    Operator::noMoreInput();
+    close();
```

Comment:
Why close is called in noMoreInput?

Reply:
Copy from here https://github.com/oap-project/velox/blob/main/velox/exec/Values.h#L36
Maybe it is not essential here?

Reply:
Removed